### ISSUE TITLE
Corner radius for apps and for active window hint is different, leaving small gap

### ISSUE THREAD

**Issue by A-Walrus** [pop-os/cosmic-epoch#1232]
Notice the small gap at the corner where the background shows through (ZOOM in on the image). The screenshot shows cosmic-term but it's the same for other cosmic apps.
![image](https://github.com/user-attachments/assets/e17ad3e4-2fce-49bf-afc9-5036f576a1ec)

**Comment by git-f0x** [comment]
The compositor doesn't know the corner radius of apps. It could be changed to match libcosmic apps, but that might conflict with apps with more square corners.
So this isn't really a bug.
There may or may not be a wayland protocol in the future for apps to report their corner radius.

**Comment by hgoohg** [comment]
> The compositor doesn't know the corner radius of apps. It could be changed to match libcosmic apps, but that might conflict with apps with more square corners. So this isn't really a bug. There may or may not be a wayland protocol in the future for apps to report their corner radius.

isn't there a way to force windows to have a certain radius? Hyprland doesn't have this issue, it allows you to set the window corner radius to your liking, and the active hint size as well (as shown in the pictures), the hint will always follow the radius you've set and the window will always draw its elements without clipping through the rounded corners, regardless if the app has sharp or round corners ootb (as shown below).

![Image](https://github.com/user-attachments/assets/2568c3db-8a50-4723-83fa-11febb3e6161)

![Image](https://github.com/user-attachments/assets/c25995dc-dfa6-4e8a-a56a-ca1f8b6d6fda)
![Image](https://github.com/user-attachments/assets/3da10ac7-8cd6-4c2d-b9a1-6079ebbbb359)

**Comment by hgoohg** [comment]
> Does it really work for all apps for any radius value, without breaking their interface (e.g. setting the radius to 64)?



my bad, youre actually right, elements do clip if you increase radius that much.

![Image](https://github.com/user-attachments/assets/3669dc5b-cd76-4a09-afe7-94e37cee6f1f)

tho i can't think of a case where any app would go so extreme on round corners, gtk4 apps tend to have the smoothest corners and theyre rarely above 30, i think hyprland gets away with doing this because it doesn't draw window decorations (such as decoration borders, buttons, title bars,etc. from themes) and only uses icon themes and the colour themes.

**Comment by elvishcraftsman** [comment]
This issue bugs me as well. I'm seeing it even on native COSMIC apps, which seem to respond to the "square" setting in Desktop > Appearance > Style. If it followed that setting, it would at least look fine on COSMIC apps. In fact, if it allowed a larger border radius when the "Slightly round" option is chosen, it might even work fairly well for GTK apps on that setting.

**Comment by Ryaja2** [comment]
Could the default outline at least be adjusted to the native cosmic windows radius?

**Comment by crash3cho** [comment]
I think as long as the maximum allowed radius was something sane the 'hyperland approach' of cropping the window would be most consistent between app types. I've used hyperland for a while and can't remember the corners ever being a problem - they were very pretty.

**Comment by Ryaja2** [comment]
Ya, if the enforced radius is small enough it shouldn't ruin any apps, it would also add visual continuity.

**Comment by jxqu3** [comment]
I agree, the hyprland approach seems like the best option. It does look like a glitch/unintended and may confuse people into thinking it's broken.

**Comment by git-f0x** [comment]
It's already fixed for libcosmic apps, and there's an upstream wayland-protocols MR. 
There will also eventually be an option to crop windows to match libcosmic radius.

**Comment by LiamillionSS** [comment]
Resolved alongside https://github.com/pop-os/cosmic-comp/issues/1713

**Comment by LiamillionSS** [comment]
Bump for resolve
